From 036d334d09bf6c585c7636f89a2ecb4f3afda408 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Sat, 11 Jan 2025 10:48:16 +0200
Subject: [PATCH] [SW-205447] gaudi2: add eeprom agent app modules

In HL288 board revision 0, second stage VR controller uses different
i2c address compared to HL225. But the address remains the same
between HL338 and HL325. Therefore, we can't rely on 'is_card_pcie'
check, but have to read the eeprom to identify that the card is HL288
and board rev is 0.

To satify this, function to read the eeprom is added to g2 common_utils.
Since this file is used in hbm, pcie and serdes agents, adding eeprom
app modules to resolve build errors.

Change-Id: I499d027e02bb174e31fd153893da2ad61373cee8
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/package/agent-pcie/agent-pcie.mk b/package/agent-pcie/agent-pcie.mk
index 055c695..25e0898 100644
--- a/package/agent-pcie/agent-pcie.mk
+++ b/package/agent-pcie/agent-pcie.mk
@@ -1,7 +1,7 @@
 AGENT_PCIE_SITE = pcie
 AGENT_PCIE_APP_NAME = pcie
 AGENT_PCIE_DEPENDENCIES = host-cmake host-lzop host-lz4
-AGENT_PCIE_APP_MODULES = boot_info i2c-indirect mbedtls soc_init hl_shell mctp pldm
+AGENT_PCIE_APP_MODULES = eeprom boot_info i2c-indirect mbedtls soc_init hl_shell mctp pldm
 
 AGENT_SRC_DIR=$(EMBEDDED_DIR)/zephyr-plugins
 AGENT_MARGIN_TOOL_SRC_DIR=$(AGENT_SRC_DIR)/margin-tools
diff --git a/package/agent-serdes/agent-serdes.mk b/package/agent-serdes/agent-serdes.mk
index feaf462..c5e693b 100644
--- a/package/agent-serdes/agent-serdes.mk
+++ b/package/agent-serdes/agent-serdes.mk
@@ -1,7 +1,7 @@
 AGENT_SERDES_SITE := agent-serdes
 AGENT_SERDES_APP_NAME = serdes
 AGENT_SERDES_DEPENDENCIES = host-cmake host-lzop host-lz4
-AGENT_SERDES_APP_MODULES = boot_info fdt i2c-indirect mbedtls soc_init hl_shell mctp pldm
+AGENT_SERDES_APP_MODULES = eeprom boot_info fdt i2c-indirect mbedtls soc_init hl_shell mctp pldm
 AGENT_SRCDIR=$(EMBEDDED_DIR)/zephyr-plugins
 AGENT_MARGIN_TOOL_SRC_DIR=$(AGENT_SRC_DIR)/margin-tools
 AGENT_SERDES_SRCDIR=$(AGENT_MARGIN_TOOL_SRC_DIR)/serdes
